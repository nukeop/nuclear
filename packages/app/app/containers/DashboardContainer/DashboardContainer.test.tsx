
import { waitFor } from '@testing-library/react';
import { buildStoreState } from '../../../test/storeBuilders';
import { mountedComponentFactory, setupI18Next } from '../../../test/testUtils';
describe('Dashboard container', () => {
  beforeAll(() => {
    setupI18Next();
  });

  it('should display the best new music page of the dashboard', () => {
    const { component } = mountComponent();
    expect(component.asFragment()).toMatchSnapshot();
  });

  it('should go to best new album review after clicking it', () => {
    const { component } = mountComponent();

    waitFor(() => component.getByText(/test title 2/i).click());

    expect(component.queryByText(/test review 2/i)).not.toBeNull();
    expect(component.asFragment()).toMatchSnapshot();
  });

  it('should go to best new track review after clicking it', () => {
    const { component } = mountComponent();

    waitFor(() => component.getByText(/test track title 2/i).click());

    expect(component.queryByText(/track review 2/i)).not.toBeNull();
    expect(component.asFragment()).toMatchSnapshot();
  });

  it('should add/remove a best new track to favorites after clicking its star', async () => {
    const { component, store } = mountComponent();

    const addOrRemove = async () => waitFor(
      () => component
        .getByTestId('favorite-icon-test track artist 1-test track title 1')
        .click()
    );

    await addOrRemove();

    let state = store.getState();
    expect(state.favorites.tracks).toEqual([
      expect.objectContaining({
        artist: 'test track artist 1',
        name: 'test track title 1'
      })
    ]);

    await addOrRemove();
    state = store.getState();

    expect(state.favorites.tracks).toEqual([]);
  });

  it('should display top tracks after going to top tracks tab', async () => {
    const { component } = mountComponent();

    await waitFor(() => component.getByText(/top tracks/i).click());

    expect(component.asFragment()).toMatchSnapshot();
  });

  it('should display genres after going to genres tab', async () => {
    const { component } = mountComponent();

    await waitFor(() => component.getByText(/genres/i).click());

    expect(component.asFragment()).toMatchSnapshot();
  });

  it('should add all top tracks to the queue', async () => {
    const { component, store } = mountComponent();

    await waitFor(() => component.getByText(/top tracks/i).click());
    await waitFor(() => component.getByText(/add all/i).click());

    const state = store.getState();
    expect(state.queue.queueItems).toEqual([
      expect.objectContaining({
        artist: 'top track artist 1',
        name: 'top track 1',
        thumbnail: 'top track thumbnail 1'
      }),
      expect.objectContaining({
        artist: 'top track artist 2',
        name: 'top track 2',
        thumbnail: 'top track thumbnail 2'
      })
    ]);
  });

  const mountComponent = mountedComponentFactory(
    ['/dashboard'],
    buildStoreState()
      .withDashboard()
      .withPlugins()
      .withConnectivity()
      .build()
  );
});
