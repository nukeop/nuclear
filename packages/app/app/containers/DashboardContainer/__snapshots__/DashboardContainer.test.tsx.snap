// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard container should display genres after going to genres tab 1`] = `
<DocumentFragment>
  <div
    class="main_layout_container"
  >
    <div
      class="dashboard"
    >
      <div
        class="dashboard_tabs"
      >
        <div
          class="ui pointing secondary menu"
        >
          <a
            class="item"
          >
            Best new music
          </a>
          <a
            class="item"
          >
            Top Tracks
          </a>
          <a
            class="active item"
          >
            Genres
          </a>
        </div>
        <div
          class="ui segment active tab"
        >
          <div
            class="genre_tab_container"
          >
            <div
              class="genre_container"
            >
              <div
                class="genre_overlay"
              >
                <div
                  class="smooth_image_wrapper"
                >
                  <img
                    class="image"
                    src="https://picsum.photos/256?random=0"
                  />
                  <img
                    class="placeholder"
                    src="test-file-stub"
                    style="opacity: 1;"
                  />
                </div>
              </div>
              <div
                class="genre_name"
              >
                <div
                  class="svg_icon"
                >
                  test-file-stub
                </div>
                rock
              </div>
            </div>
            <div
              class="genre_container"
            >
              <div
                class="genre_overlay"
              >
                <div
                  class="smooth_image_wrapper"
                >
                  <img
                    class="image"
                    src="https://picsum.photos/256?random=1"
                  />
                  <img
                    class="placeholder"
                    src="test-file-stub"
                    style="opacity: 1;"
                  />
                </div>
              </div>
              <div
                class="genre_name"
              >
                <div
                  class="svg_icon"
                >
                  test-file-stub
                </div>
                pop
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Dashboard container should display the best new music page of the dashboard 1`] = `
<DocumentFragment>
  <div
    class="main_layout_container"
  >
    <div
      class="dashboard"
    >
      <div
        class="dashboard_tabs"
      >
        <div
          class="ui pointing secondary menu"
        >
          <a
            class="active item"
          >
            Best new music
          </a>
          <a
            class="item"
          >
            Top Tracks
          </a>
          <a
            class="item"
          >
            Genres
          </a>
        </div>
        <div
          class="ui segment active tab best_new_music_tab_pane"
        >
          <div
            class="best_new_music_menu"
          >
            <div
              class="ui large vertical menu"
            >
              <div
                class="link item best_new_music_menu_header"
              >
                Best new albums
              </div>
              <div
                class="link item"
              >
                <div
                  class="best_new_music_card"
                >
                  <div
                    class="card_thumbnail"
                  >
                    <img
                      alt="test title 1"
                      src="test thumbnail 1"
                    />
                  </div>
                  <div
                    class="card_info"
                  >
                    <div
                      class="card_title_row"
                    >
                      <div
                        class="card_title"
                      >
                        test title 1
                      </div>
                    </div>
                    <div
                      class="card_artist"
                    >
                      test artist 1
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="link item"
              >
                <div
                  class="best_new_music_card"
                >
                  <div
                    class="card_thumbnail"
                  >
                    <img
                      alt="test title 2"
                      src="test thumbnail 2"
                    />
                  </div>
                  <div
                    class="card_info"
                  >
                    <div
                      class="card_title_row"
                    >
                      <div
                        class="card_title"
                      >
                        test title 2
                      </div>
                    </div>
                    <div
                      class="card_artist"
                    >
                      test artist 2
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="link item best_new_music_menu_header"
              >
                Best new tracks
              </div>
              <div
                class="link item"
              >
                <div
                  class="best_new_music_card"
                >
                  <div
                    class="card_thumbnail"
                  >
                    <img
                      alt="test track title 1"
                      src="test track thumbnail 1"
                    />
                  </div>
                  <div
                    class="card_info"
                  >
                    <div
                      class="card_title_row"
                    >
                      <div
                        class="card_title"
                      >
                        test track title 1
                      </div>
                      <i
                        aria-hidden="true"
                        class="star outline large icon card_favorite"
                        data-testid="favorite-icon-test track artist 1-test track title 1"
                      />
                    </div>
                    <div
                      class="card_artist"
                    >
                      test track artist 1
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="link item"
              >
                <div
                  class="best_new_music_card"
                >
                  <div
                    class="card_thumbnail"
                  >
                    <img
                      alt="test track title 2"
                      src="test track thumbnail 2"
                    />
                  </div>
                  <div
                    class="card_info"
                  >
                    <div
                      class="card_title_row"
                    >
                      <div
                        class="card_title"
                      >
                        test track title 2
                      </div>
                      <i
                        aria-hidden="true"
                        class="star outline large icon card_favorite"
                        data-testid="favorite-icon-test track artist 2-test track title 2"
                      />
                    </div>
                    <div
                      class="card_artist"
                    >
                      test track artist 2
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Dashboard container should display top tracks after going to top tracks tab 1`] = `
<DocumentFragment>
  <div
    class="main_layout_container"
  >
    <div
      class="dashboard"
    >
      <div
        class="dashboard_tabs"
      >
        <div
          class="ui pointing secondary menu"
        >
          <a
            class="item"
          >
            Best new music
          </a>
          <a
            class="active item"
          >
            Top Tracks
          </a>
          <a
            class="item"
          >
            Genres
          </a>
        </div>
        <div
          class="ui segment active tab"
        >
          <div
            class="popular_tracks_container tracks_container"
          >
            <div
              class="popular_tracks_header"
            >
              <h3>
                Top Tracks from Deezer
              </h3>
              <a
                aria-label="Add all"
                class="ui primary button add_button"
                href="#"
                role="button"
              >
                <span
                  aria-hidden="true"
                  class="fa fa-plus"
                />
                 Add all
              </a>
            </div>
            <table>
              <thead>
                <tr>
                  <th>
                    <span
                      aria-hidden="true"
                      class="fa fa-image"
                    />
                  </th>
                  <th>
                    Artist
                  </th>
                  <th>
                    Title
                  </th>
                </tr>
                <tr />
              </thead>
              <tbody>
                <tr
                  class="track"
                >
                  <td
                    class="track_thumbnail"
                  >
                    <img
                      src="top track thumbnail 1"
                    />
                  </td>
                  <td
                    class="track_artist"
                  >
                    top track artist 1
                  </td>
                  <td
                    class="track_name"
                  >
                    top track 1
                  </td>
                </tr>
                <tr
                  class="track"
                >
                  <td
                    class="track_thumbnail"
                  >
                    <img
                      src="top track thumbnail 2"
                    />
                  </td>
                  <td
                    class="track_artist"
                  >
                    top track artist 2
                  </td>
                  <td
                    class="track_name"
                  >
                    top track 2
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Dashboard container should go to best new album review after clicking it 1`] = `
<DocumentFragment>
  <div
    class="main_layout_container"
  >
    <div
      class="dashboard"
    >
      <div
        class="dashboard_tabs"
      >
        <div
          class="ui pointing secondary menu"
        >
          <a
            class="active item"
          >
            Best new music
          </a>
          <a
            class="item"
          >
            Top Tracks
          </a>
          <a
            class="item"
          >
            Genres
          </a>
        </div>
        <div
          class="ui segment active tab best_new_music_tab_pane"
        >
          <div
            class="best_new_music_menu"
          >
            <div
              class="ui large vertical menu"
            >
              <div
                class="link item best_new_music_menu_header"
              >
                Best new albums
              </div>
              <div
                class="link item"
              >
                <div
                  class="best_new_music_card"
                >
                  <div
                    class="card_thumbnail"
                  >
                    <img
                      alt="test title 1"
                      src="test thumbnail 1"
                    />
                  </div>
                  <div
                    class="card_info"
                  >
                    <div
                      class="card_title_row"
                    >
                      <div
                        class="card_title"
                      >
                        test title 1
                      </div>
                    </div>
                    <div
                      class="card_artist"
                    >
                      test artist 1
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="link item"
              >
                <div
                  class="best_new_music_card"
                >
                  <div
                    class="card_thumbnail"
                  >
                    <img
                      alt="test title 2"
                      src="test thumbnail 2"
                    />
                  </div>
                  <div
                    class="card_info"
                  >
                    <div
                      class="card_title_row"
                    >
                      <div
                        class="card_title"
                      >
                        test title 2
                      </div>
                    </div>
                    <div
                      class="card_artist"
                    >
                      test artist 2
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="link item best_new_music_menu_header"
              >
                Best new tracks
              </div>
              <div
                class="link item"
              >
                <div
                  class="best_new_music_card"
                >
                  <div
                    class="card_thumbnail"
                  >
                    <img
                      alt="test track title 1"
                      src="test track thumbnail 1"
                    />
                  </div>
                  <div
                    class="card_info"
                  >
                    <div
                      class="card_title_row"
                    >
                      <div
                        class="card_title"
                      >
                        test track title 1
                      </div>
                      <i
                        aria-hidden="true"
                        class="star outline large icon card_favorite"
                        data-testid="favorite-icon-test track artist 1-test track title 1"
                      />
                    </div>
                    <div
                      class="card_artist"
                    >
                      test track artist 1
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="link item"
              >
                <div
                  class="best_new_music_card"
                >
                  <div
                    class="card_thumbnail"
                  >
                    <img
                      alt="test track title 2"
                      src="test track thumbnail 2"
                    />
                  </div>
                  <div
                    class="card_info"
                  >
                    <div
                      class="card_title_row"
                    >
                      <div
                        class="card_title"
                      >
                        test track title 2
                      </div>
                      <i
                        aria-hidden="true"
                        class="star outline large icon card_favorite"
                        data-testid="favorite-icon-test track artist 2-test track title 2"
                      />
                    </div>
                    <div
                      class="card_artist"
                    >
                      test track artist 2
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="best_new_music_content"
          >
            <div
              class="review_header"
            >
              <div
                class="thumbnail"
              >
                <img
                  alt="test title 2"
                  src="test thumbnail 2"
                />
              </div>
              <div
                class="review_headings"
              >
                <div
                  class="artist"
                >
                  test artist 2
                </div>
                <div
                  class="title"
                >
                  test title 2
                </div>
              </div>
              <div
                class="score"
              >
                <div
                  class="score_box"
                >
                  8.0
                </div>
              </div>
            </div>
            <p
              class="paragraph"
            >
              test review 2
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Dashboard container should go to best new track review after clicking it 1`] = `
<DocumentFragment>
  <div
    class="main_layout_container"
  >
    <div
      class="dashboard"
    >
      <div
        class="dashboard_tabs"
      >
        <div
          class="ui pointing secondary menu"
        >
          <a
            class="active item"
          >
            Best new music
          </a>
          <a
            class="item"
          >
            Top Tracks
          </a>
          <a
            class="item"
          >
            Genres
          </a>
        </div>
        <div
          class="ui segment active tab best_new_music_tab_pane"
        >
          <div
            class="best_new_music_menu"
          >
            <div
              class="ui large vertical menu"
            >
              <div
                class="link item best_new_music_menu_header"
              >
                Best new albums
              </div>
              <div
                class="link item"
              >
                <div
                  class="best_new_music_card"
                >
                  <div
                    class="card_thumbnail"
                  >
                    <img
                      alt="test title 1"
                      src="test thumbnail 1"
                    />
                  </div>
                  <div
                    class="card_info"
                  >
                    <div
                      class="card_title_row"
                    >
                      <div
                        class="card_title"
                      >
                        test title 1
                      </div>
                    </div>
                    <div
                      class="card_artist"
                    >
                      test artist 1
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="link item"
              >
                <div
                  class="best_new_music_card"
                >
                  <div
                    class="card_thumbnail"
                  >
                    <img
                      alt="test title 2"
                      src="test thumbnail 2"
                    />
                  </div>
                  <div
                    class="card_info"
                  >
                    <div
                      class="card_title_row"
                    >
                      <div
                        class="card_title"
                      >
                        test title 2
                      </div>
                    </div>
                    <div
                      class="card_artist"
                    >
                      test artist 2
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="link item best_new_music_menu_header"
              >
                Best new tracks
              </div>
              <div
                class="link item"
              >
                <div
                  class="best_new_music_card"
                >
                  <div
                    class="card_thumbnail"
                  >
                    <img
                      alt="test track title 1"
                      src="test track thumbnail 1"
                    />
                  </div>
                  <div
                    class="card_info"
                  >
                    <div
                      class="card_title_row"
                    >
                      <div
                        class="card_title"
                      >
                        test track title 1
                      </div>
                      <i
                        aria-hidden="true"
                        class="star outline large icon card_favorite"
                        data-testid="favorite-icon-test track artist 1-test track title 1"
                      />
                    </div>
                    <div
                      class="card_artist"
                    >
                      test track artist 1
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="link item"
              >
                <div
                  class="best_new_music_card"
                >
                  <div
                    class="card_thumbnail"
                  >
                    <img
                      alt="test track title 2"
                      src="test track thumbnail 2"
                    />
                  </div>
                  <div
                    class="card_info"
                  >
                    <div
                      class="card_title_row"
                    >
                      <div
                        class="card_title"
                      >
                        test track title 2
                      </div>
                      <i
                        aria-hidden="true"
                        class="star outline large icon card_favorite"
                        data-testid="favorite-icon-test track artist 2-test track title 2"
                      />
                    </div>
                    <div
                      class="card_artist"
                    >
                      test track artist 2
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="best_new_music_content"
          >
            <div
              class="review_header"
            >
              <div
                class="thumbnail"
              >
                <img
                  alt="test track title 2"
                  src="test track thumbnail 2"
                />
              </div>
              <div
                class="review_headings"
              >
                <div
                  class="artist"
                >
                  test track artist 2
                </div>
                <div
                  class="title"
                >
                  test track title 2
                </div>
              </div>
            </div>
            <p
              class="paragraph"
            >
              track review 2
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
